pkg/model/model.go

```diff
@@ -462,6 +462,14 @@ func (m *Model) GetTypeDefs() ([]*TypeDef, error) {
 		return tdefs[i].Names.Camel < tdefs[j].Names.Camel
 	})
 	m.processNestedFieldTypeDefs(tdefs)
+
+	// Add custom type definitions for custom shapes
+	var err error
+	tdefs, err = m.AddCustomTypeDefinitions(tdefs)
+	if err != nil {
+		return nil, fmt.Errorf("error adding custom type definitions: %w", err)
+	}
```

Comment:
maybe this function can just return the customTypes and we'll append them.

```suggestion
	customTypes, err := m.AddCustomTypeDefinitions(tdefs)
	if err != nil {
		return nil, fmt.Errorf("error adding custom type definitions: %w", err)
	}
```